rbmc-cfam-daemon/src/local_bmc.cpp

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "local_bmc.hpp"
+
+#include <phosphor-logging/lg2.hpp>
+#include <sdbusplus/async.hpp>
+#include <xyz/openbmc_project/State/BMC/Redundancy/common.hpp>
+#include <xyz/openbmc_project/State/BMC/client.hpp>
+
+constexpr uint8_t apiVersion{0x01};
+
+sdbusplus::async::task<> LocalBMC::start()
+{
+    writeApiVersion();
+    writeFWVersion();
+    writeBMCPosition();
+    writeProvisioned();
+    writeSiblingCommsNotOK();
+
+    services = std::make_unique<Services>(
+        ctx, [this](auto state) { bmcStateChanged(state); },
+        [this](auto role) { roleChanged(role); },
+        [this](auto enabled) { redEnabledChanged(enabled); });
+
+    co_await writeRedundancyProps();
+    co_await writeBMCState();
+
+    ctx.spawn(watchHeartBeat());
+
+    co_return;
+}
+
+sdbusplus::async::task<> LocalBMC::watchHeartBeat()
+{
+    using namespace sdbusplus::bus::match;
+    using Redundancy =
+        sdbusplus::common::xyz::openbmc_project::state::bmc::Redundancy;
+
+    sdbusplus::async::match match(ctx, rules::interface(Redundancy::interface) +
+                                           rules::member("Heartbeat"));
+
+    while (!ctx.stop_requested())
+    {
+        co_await match.next<>();
+
+        cfam.incHeartbeat();
+    }
+
+    co_return;
+}
+
+void LocalBMC::writeApiVersion()
+{
+    cfam.writeApiVersion(apiVersion);
+}
+
+void LocalBMC::writeFWVersion()
+{
+    const std::filesystem::path osRelease{"/etc/os-release"};
+
+    auto versionID = services->getVersionID(osRelease);
+
+    // Write the hash to the CFAM
+    uint32_t version = std::hash<std::string>{}(versionID);
+    cfam.writeFWVersion(version);
+}
+
+void LocalBMC::writeBMCPosition()
+{
+    cfam.writeBMCPosition(services->getBMCPosition());
+}
+
+void LocalBMC::writeSiblingCommsNotOK()
+{
+    cfam.writeSiblingCommsOK(false);
```

Comment:
Do we need to add any TODO? I think it will be set based on some check, right?

Reply:
This call here initializes it to false on startup, and then when sibling bmc class is added it can then be set based on the results of being able to read it:

in the while loop in application.cpp:
```
        if (siblingBMC)
        {
            siblingBMC->read();
            localBMC.setSiblingCommsOK(siblingBMC->ok());
        }

```